File: Programmers/Level 2/N개의 최소공배수.py
def least_common_multiple(a, b):
    div = 2
    com_mul = 1
    while (div <= min(a, b)):
        if a % div == 0 and b % div == 0:
            a = a // div
            b = b // div
            com_mul *= div
            div = 2
        else:
            div += 1
    com_mul *= a * b
    return com_mul
            
def solution(arr):
    answer = 1
    a = arr[0]
    for i in range(1, len(arr)):
        b = arr[i]
        a = least_common_multiple(a, b)
        
    answer = a
    return answer
  
''' gcd 함수 사용한 사람 있음'''

File: Programmers/JS/N개의 최소공배수.js
function gcd(num1, num2){
    while(num2){
        let temp = num1 % num2
        num1 = num2
        num2 = temp
    }
    return num1
}

function lcm(num1, num2){
    return (num1 * num2 / gcd(num1, num2))
}
function solution(arr) {
    var answer = 0;
    let num = arr[0]
    arr.sort((a, b)=> a - b)
    for(let i = 1; i < arr.length; i++){
        num = lcm(num, arr[i])
    }
    return num;
}
